eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -492,9 +493,17 @@ class Peer(nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: Actor
     case Event(_: BadMessage, _) => stay // we got disconnected while syncing
   }
 
+  /**
+    * The transition INSTANTIATING -> DISCONNECTED happens in 2 scenarios
+    *   - Manual connection to a new peer: then when(DISCONNECTED) we expect a Peer.Connect from the switchboard
+    *   - Eclair restart: The switchboard creates the peers and sends Init and then Peer.Reconnect to trigger reconnection attempts
+    *
+    * So when we see this transition we NO-OP because we don't want to start a Reconnect timer but the peer will receive the trigger
+    * (Connect/Reconnect) messages from the switchboard.
+    */
   onTransition {
-    case INSTANTIATING -> DISCONNECTED if nodeParams.autoReconnect && nextStateData.address_opt.isDefined => self ! Reconnect // we reconnect right away if we just started the peer
-    case _ -> DISCONNECTED if nodeParams.autoReconnect => setTimer(RECONNECT_TIMER, Reconnect, 1 second, repeat = false)
+    case INSTANTIATING -> DISCONNECTED => ()
+    case _ -> DISCONNECTED if nodeParams.autoReconnect => setTimer(RECONNECT_TIMER, Reconnect, Random.nextInt(5000).millis, repeat = false) // we add some randomization to not have peers reconnect to each other exactly at the same time
```

Comment:
We could use a configuration setting instead of hardcoding 5000 milliseconds here

Reply:
Isn't it a bit too much have this configurable? The randomization here prevents from starting the reconnection attempts exactly at the same time and we just want to make sure it happens as soon as possible but not exactly synchronized with the remote peer (if the remote is using eclair as well).

Reply:
I think it doesn't hurt, will add.

Reply:
done with b8a056c